pandas/core/indexes/multi.py

```diff
@@ -235,16 +235,28 @@ def set_levels(self, levels, level=None, inplace=False,
                    labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
                    names=[u'foo', u'bar'])
         """
-        if level is not None and not is_list_like(level):
-            if not is_list_like(levels):
-                raise TypeError("Levels must be list-like")
-            if is_list_like(levels[0]):
-                raise TypeError("Levels must be list-like")
+
```

Comment:
this validation is quite similar to ``.from_arrays``.  and should coordinate with changes in #14823  (cc @groutr)